hadoop-ozone/tools/src/main/java/org/apache/hadoop/ozone/repair/om/FSORepairTool.java

```diff
@@ -0,0 +1,757 @@
+        OM_KEY_PREFIX +
+        bucket.getObjectID() +
+        OM_KEY_PREFIX +
+        object.getObjectID();
+  }
+
+  /**
+   *
+   * @param fileOrDirKey The key of a file or directory in RocksDB.
+   * @return true if the entry's parent is in the reachable table.
+   */
+  protected boolean isReachable(String fileOrDirKey) throws IOException {
+    byte[] reachableParentKey = buildReachableParentKey(fileOrDirKey).getBytes(StandardCharsets.UTF_8);
+
+    return reachableDB.get(reachableCFHandle, reachableParentKey, REACHABLE_TABLE) != null;
+  }
+
+  /**
+   * Build an entry in the reachable table for the current object's parent
+   * object. The object could be a file or directory.
+   */
+  private static String buildReachableParentKey(String fileOrDirKey) {
+    String[] keyParts = fileOrDirKey.split(OM_KEY_PREFIX);
+    // Should be /volID/bucketID/parentID/name
+    // The first part will be blank since key begins with a slash.
+    Preconditions.assertTrue(keyParts.length >= 4);
+    String volumeID = keyParts[1];
+    String bucketID = keyParts[2];
+    String parentID = keyParts[3];
+
+    return OM_KEY_PREFIX +
+        volumeID +
+        OM_KEY_PREFIX +
+        bucketID +
+        OM_KEY_PREFIX +
+        parentID;
+  }
+
+  private void openReachableDB() throws IOException {
+    File reachableDBFile = new File(new File(omDBPath).getParentFile(), "reachable.db");
+    System.out.println("Creating database of reachable directories at " + reachableDBFile);
+    // Delete the DB from the last run if it exists.
+    if (reachableDBFile.exists()) {
+      FileUtils.deleteDirectory(reachableDBFile);
+    }
+    reachableDBPath = reachableDBFile.toString();
+    reachableDB = buildReachableRocksDB(reachableDBFile);
+  }
+
+  private RocksDatabase buildReachableRocksDB(File reachableDBFile) throws IOException {
+    DBProfile profile = new OzoneConfiguration().getEnum(HDDS_DB_PROFILE, HDDS_DEFAULT_DB_PROFILE);
+    Set<TableConfig> tableConfigs = new HashSet<>();
+
+    try {
+      tableConfigs.add(new TableConfig("default", profile.getColumnFamilyOptions()));
+
+      return RocksDatabase.open(reachableDBFile,
+          profile.getDBOptions(),
+          new ManagedWriteOptions(),
+          tableConfigs, false);
+    } finally {
+      for (TableConfig config : tableConfigs) {
+        config.close();
+      }
+    }
+  }
+
+  private void closeReachableDB() {
+    if (reachableDB != null) {
+      reachableDB.close();
+    }
+  }
+
+  private void dropReachableTableIfExists() throws IOException {
+    try {
+      List<byte[]> availableCFs = reachableDB.listColumnFamiliesEmptyOptions(reachableDBPath);
+      boolean cfFound = false;
+      for (byte[] cfNameBytes: availableCFs) {
+        if (new String(cfNameBytes, UTF_8).equals(new String(REACHABLE_TABLE_BYTES, UTF_8))) {
+          cfFound = true;
+          break;
+        }
+      }
+
+      if (cfFound) {
+        reachableDB.dropColumnFamily(reachableCFHandle);
```

Comment:
We can follow common mechanism to add table column family, currently to manage this db have a lot of code, which can be simplified.
refer code in below for same,
https://github.com/apache/ozone/pull/7465/files#diff-e28b4ee9f893afcd5d1a5aabd780cf947b7bde94b91aafcb5e8308120a71690bR244